src/game_inventory.cpp

```diff
@@ -1376,3 +1383,184 @@ void game_menus::inv::swap_letters( player &p )
+                     !pa.has_bionic( bid->upgraded_bionic ) &&
+                     it->is_upgrade() ) {
+                return "NO BASE VERSION INSTALLED";
+            }
+            else if( std::any_of( bid->available_upgrades.begin(),
+                     bid->available_upgrades.end(),
+                     std::bind( &player::has_bionic, &pa,
+                     std::placeholders::_1 ) ) ) {
+                return "SUPERIOR VERSION INSTALLED";
+            }
+            else if( pa.is_npc() && !bid->npc_usable ) {
+                return "CBM NOT COMPATIBLE WITH PATIENT";
+            }
+
+            return std::string();
+        }
+
+    protected:
+        player &p;
+        player &pa;
+
+    private:
+        // Returns a formatted string of how long the operation will take.
+        std::string get_operation_duration( const item_location & loc ) {
+            const item *it = loc.get_item();
+            const itype *itemtype = it->type;
+            const int difficulty = itemtype->bionic->difficulty;
+
+            // 20 minutes per bionic difficulty.
+            int hours = difficulty / 3;
+            int minutes = (difficulty % 3) * 20;
+            std::string minutes_string = minutes > 0
+                                        ? string_format( _( "%i minutes" ), minutes )
+                                        : std::string();
+
+            if( hours > 0 ) {
+                std::string hours_string = hours >= 2
+                                        ? string_format( _( "%i hours" ), hours )
+                                        : string_format( _( "%i hour" ), hours );
+
+                if( minutes > 0 ) {
+                    return string_format( _( "%s, %s" ), hours_string, minutes_string );
+                }
+                else {
+                    return hours_string;
+                }
+            }
+            else {
+                return minutes_string;
+            }
+        }
+
+        // Failure chance for bionic install. Combines multiple other functions together.
+        std::string get_failure_chance( const item_location & loc ) {
+            const item *it = loc.get_item();
+            const itype *itemtype = it->type;
+            const int difficulty = itemtype->bionic->difficulty;
+            int chance_of_failure = 100;
+            player & installer = p;
+
+            const int adjusted_skill = installer.bionics_adjusted_skill( skill_firstaid,
+                             skill_computer,
+                             skill_electronics,
+                             -1 );
+
+            if( ( get_option < bool > ( "SAFE_AUTODOC" ) ) ||
+                g->u.has_trait( trait_id( "DEBUG_BIONICS" ) ) ) {
+                chance_of_failure = 0;
+            }
+            else {
+                float skill_difficulty_parameter = static_cast<float>( adjusted_skill /
+                                           ( 4.0 * difficulty ) );
```

Comment:
```suggestion
                                           4.0 * difficulty );
```

Reply:
Can't do this change. It changes the order of operations and messes up the formula.

Reply:
How's that? `x / ( 4 * y ) = x / 4 * y` .

Reply:
It should be `x / y / 4`.